src/monitor/pgautofailover--1.3--1.4.sql

```diff
@@ -0,0 +1,782 @@
+         join pgautofailover.formation using(formationid)
+   where formationid = formation_id
+     and groupid = 0
+     and goalstate in ('single', 'wait_primary', 'primary')
+     and reportedstate in ('single', 'wait_primary', 'primary');
+$$;
+
+grant execute on function pgautofailover.get_coordinator(text)
+   to autoctl_node;
+
+
+CREATE FUNCTION pgautofailover.get_most_advanced_standby
+ (
+   IN formationid       text default 'default',
+   IN groupid           int default 0,
+   OUT node_id          bigint,
+   OUT node_name        text,
+   OUT node_host        text,
+   OUT node_port        int,
+   OUT node_lsn         pg_lsn,
+   OUT node_is_primary  bool
+ )
+RETURNS SETOF record LANGUAGE SQL STRICT
+AS $$
+   select nodeid, nodename, nodehost, nodeport, reportedlsn, false
+     from pgautofailover.node
+    where formationid = $1
+      and groupid = $2
+      and reportedstate = 'report_lsn'
+ order by reportedlsn desc, health desc
+    limit 1;
+$$;
+
+grant execute on function pgautofailover.get_most_advanced_standby(text,int)
+   to autoctl_node;
+
+DROP FUNCTION IF EXISTS pgautofailover.remove_node(text, int);
+
+CREATE FUNCTION pgautofailover.remove_node
+ (
+   node_id int
+ )
+RETURNS bool LANGUAGE C STRICT SECURITY DEFINER
+AS 'MODULE_PATHNAME', $$remove_node_by_nodeid$$;
+
+comment on function pgautofailover.remove_node(int)
+        is 'remove a node from the monitor';
+
+grant execute on function pgautofailover.remove_node(int)
+   to autoctl_node;
+
+CREATE FUNCTION pgautofailover.remove_node
+ (
+   node_host text,
+   node_port int default 5432
+ )
+RETURNS bool LANGUAGE C STRICT SECURITY DEFINER
+AS 'MODULE_PATHNAME', $$remove_node_by_host$$;
+
+comment on function pgautofailover.remove_node(text,int)
+        is 'remove a node from the monitor';
+
+grant execute on function pgautofailover.remove_node(text,int)
+   to autoctl_node;
+
+CREATE FUNCTION pgautofailover.perform_promotion
+ (
+  formation_id text,
+  node_name    text
+ )
+RETURNS bool LANGUAGE C STRICT SECURITY DEFINER
+AS 'MODULE_PATHNAME', $$perform_promotion$$;
+
+comment on function pgautofailover.perform_promotion(text,text)
+        is 'manually failover from the primary to the given node';
+
+DROP FUNCTION pgautofailover.start_maintenance(text, int);
+DROP FUNCTION pgautofailover.stop_maintenance(text, int);
+
+CREATE FUNCTION pgautofailover.start_maintenance(node_id int)
+RETURNS bool LANGUAGE C STRICT SECURITY DEFINER
+AS 'MODULE_PATHNAME', $$start_maintenance$$;
+
+comment on function pgautofailover.start_maintenance(int)
+        is 'set a node in maintenance state';
+
+grant execute on function pgautofailover.start_maintenance(int)
+   to autoctl_node;
+
+CREATE FUNCTION pgautofailover.stop_maintenance(node_id int)
+RETURNS bool LANGUAGE C STRICT SECURITY DEFINER
+AS 'MODULE_PATHNAME', $$stop_maintenance$$;
+
+comment on function pgautofailover.stop_maintenance(int)
+        is 'set a node out of maintenance state';
+
```

Comment:
It's missing this line as compared to the non upgrade script
```suggestion
grant execute on function pgautofailover.stop_maintenance(int)
   to autoctl_node;
```

Reply:
Done now.